cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -916,9 +925,15 @@ TYPED_TEST(TestParquetIO, SingleColumnOptionalReadWrite) {
 }
 
 TYPED_TEST(TestParquetIO, SingleColumnOptionalDictionaryWrite) {
-  // Skip tests for BOOL as we don't create dictionaries for it.
-  if (TypeParam::type_id == ::arrow::Type::BOOL) {
-    return;
+  switch (TypeParam::type_id) {
+    // Skip tests for BOOL as we don't create dictionaries for it.
+    case ::arrow::Type::BOOL:
+    // Skip tests for HALF_FLOAT as it's not currently supported by `dictionary_encode`
+    case ::arrow::Type::HALF_FLOAT:
+      GTEST_SKIP();
```

Comment:
Let's add a skip message for clarity (though I'm not sure GTest always displays it).
```suggestion
      GTEST_SKIP() << "dictionary_encode not supported for float16" ;
```

Reply:
Also do the same for bool above.